### PR TITLE
Change each plan column into a proper form

### PR DESCRIPTION
**Pull request by sslotsky**
## Reason for change

This prevents users from using the _Get Started_ button if any of the configurable feature values are invalid.

For example, if a numeric feature has an increment of 10, and a user specifies a number that's between increments, the user should see a validation error and should be unable to proceed without fixing it.

I implemented this with the following changes:

1. Change each column into a proper form
1. Add a `name` attribute to each configurable feature input such that it matches the feature label
1. Add a hidden `planId` input in the form

This _Uses The Platform_ :tm: so that: 

1. A form submission will not occur if any fields are invalid
1. A validation error will be presented to the user for any invalid field
1. If the form is valid, we'll raise a custom event and perform analytics tracking
1. If they supplied a URL, the event's default behavior will proceed, so the form will submit to the supplied URL via GET, and data from the form inputs will be added to the query string automatically

## Testing

You'll need a product with configurable features to test with. This PR changes the environment and product ID so that the product data will include configurable features.

For this product ID, the increment for the _Jus Ensache_ feature is 10, but the user has entered 55 and tried to submit the form. Instead of submitting, a validation message is displayed.

Now try submitting a valid form. A page transition should occur, and the new URL in the address bar should contain the planID and configurable feature selections that you'd chosen, e.g. http://localhost:3333/signup?planId=235e9tww1qvd3pkyn2j4x7937u3e4&sandwich=free&juice=1

## Checklist

- [ ] [CHANGELOG](./CHANGELOG.md) updated

### CODE REVIEW

File: src/components/manifold-plan-table/manifold-plan-table.tsx [hunk 1]
```diff
@@ -420,48 +403,58 @@ export class ManifoldPlanTable {
...
-              );
-            }),
-            <div
-              class="ManifoldPlanTable__Cell ManifoldPlanTable__Cell--Body"
-              data-row-last
-              data-column-last={lastColumn}
+            <form
+              action={this.baseUrl}
+              onSubmit={this.onSubmit(plan.id)}
+              style={{ display: 'contents' }}
```
Comment (sslotsky) [inline src/components/manifold-plan-table/manifold-plan-table.tsx:409]: I suppose I should move this to the stylesheet.

Comment (dangodev) [inline src/components/manifold-plan-table/manifold-plan-table.tsx:409]: Could you move this style into maybe `ManifoldPlanTable__PlanColumn` or something in `_plan.scss`?

I don’t think a user would realistically want to override this, but it still breaks our pattern of CSS class-level specificity (we try and not inline properties as a general pattern to keep things flexible)

File: src/components/manifold-plan-table/manifold-plan-table.tsx [hunk 2]
```diff
@@ -420,48 +403,58 @@ export class ManifoldPlanTable {
...
-                data-testid="cta"
-                class="ManifoldPlanTable__Button"
-                id={`manifold-cta-plan-${plan.id}`}
-                href={this.ctaHref(plan.id)}
-                onClick={this.handleCtaClick(plan.id)}
-              >
-                {this.ctaText}
-              </a>
-            </div>,
+              <input hidden name="planId" value={plan.id} />
```
Comment (dangodev) [inline src/components/manifold-plan-table/manifold-plan-table.tsx:411]: 👍

Comment (dangodev) [reply src/components/manifold-plan-table/manifold-plan-table.tsx:411]: Oh wait this should be `type="hidden"` (though I think `hidden` may work the same)

File: src/components/manifold-plan-table/manifold-plan-table.tsx (hunk 1 above)
Comment (dangodev) [inline src/components/manifold-plan-table/manifold-plan-table.tsx:409]: Should we add `method="GET"` here? Is that always the default?

Comment (sslotsky) [reply src/components/manifold-plan-table/manifold-plan-table.tsx:409]: Absolutely

Comment (sslotsky) [reply src/components/manifold-plan-table/manifold-plan-table.tsx:409]: Yeah it's the default

**Pull request merged**